src/qt/researcher/researchermodel.cpp

```diff
@@ -306,7 +306,13 @@ QString ResearcherModel::email() const
 
 QString ResearcherModel::formatCpid() const
 {
-    return QString::fromStdString(m_researcher->Id().ToString());
+    QString text = QString::fromStdString(m_researcher->Id().ToString());
+
+    if (m_privacy_enabled) {
+        text = text.replace(QRegExp("."), "#");
```

Comment:
Regex is an overkill for this, my suggestion is below if you don't want to hardcode:
```suggestion
        text = QString("#").repeated(text.size());
```
or if you prefer to hardcode: (which is my preference as it is constant whether the Id is an investor or not, although the difference in UI fields might still leak this data.)
```suggestion
        text = "################################";
```

Reply:
Whatever. :|.